eks/managing_user_access.md

```diff
@@ -0,0 +1,232 @@
+In the following example we will create a role which will have limited access to a `develop` namespace and then map it to an IAM user. You will need to be a cluster administrator as detailed above to proceed with the following steps.
+
+First we will need to add the user's ARN to the aws-auth configmap in the kube-system namespace.
+
+1. First we edit the `aws-auth` configmap:
+- Create a local copy of the aws-auth configmap: `kubectl get configmap -n kube-system aws-auth -o yaml > aws-auth.yaml`
+
+- Then add the user details under `mapUsers` in the configmap's data:
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: aws-auth
+  namespace: kube-system
+data:
+  mapUsers: |
+    - userarn: arn:aws:iam::<account>:user/<dev-user>
+      username: <dev-user>
+```
+
+- Apply your changes:
+`kubectl apply -f aws-auth.yaml`
+
+
+2. Now we'll create a `role` that may only read pod data in the `develop` namespace.
+- add the following to a file called `read-role.yaml`:
+```
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  namespace: develop
+  name: pod-reader
+rules:
+- apiGroups: [""] # "" indicates the core API group
+  resources: ["pods"]
+  verbs: ["list","get","watch"]
+- apiGroups: ["extensions","apps"]
+  resources: ["deployments"]
+  verbs: ["get", "list", "watch"]
+```
+
+- and then apply the role to your cluster:
+`kubectl apply -f read-role.yaml`
+
+
+3. Now we will create a `role-binding` on our user's ARN and the pod-reader role in the cluster to define their access level.
+- add the following to a file called `read-role-binding.yaml`
+```
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: read-pods
+  namespace: rbac-test
+subjects:
+- kind: User
+  name: rbac-user
+  apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: Role
+  name: pod-reader
+  apiGroup: rbac.authorization.k8s.io
+```
+
+- and then apply the role-binding to your cluster:
+`kubectl apply -f read-role-binding.yaml`
+
+4. Finally, to grant access to the cluster, you will need to whitelist the user's IP address.
+- As with admin users, we will need to add the user's IP to your terrafrom.tfvars and `terraform apply`
+
+Now your user will be able to access the cluster but will only be able to view pod resources in the develop namespace.
+
+For more details on managing permissions, you may read Google's [RBAC] documentation.
+
+
+## Providing Access with IAM Groups
+You may not wish to repeat the process above for each user you wish to add. Many users will require the same access levels. We can limit the repitition by assigning users to IAM groups.
+Similary to users, groups may also be mapped to roles to provide access management.
+
+1. First create an IAM role:
+```
+POLICY=$(echo -n '{"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":{"AWS":"arn:aws:iam::'; echo -n "$ACCOUNT_ID"; echo -n ':root"},"Action":"sts:AssumeRole","Condition":{}}]}')
```

Comment:
Can we be sure that $ACCOUNT_ID is set here and has the correct value?

Reply:
made this a bit more explicit